zarr/indexing.py

```diff
@@ -822,3 +822,62 @@ def pop_fields(selection):
         selection = tuple(s for s in selection if not isinstance(s, str))
         selection = selection[0] if len(selection) == 1 else selection
     return fields, selection
+
+
+def selection_size(selection, arr):
+    if len(selection) > len(arr.shape):
+        raise ValueError(f'dimensions in selection cant be greater than dimensions or array: {len(selection)} > {len(arr.shape)}')
+    selection_shape = []
+    for i, size in arr.shape:
+        selection_slice = selection[i] if i < len(selection) else None
+        if selection_slice:
+            selection_slice_size = len(range(*selection_slice.indices(len(arr))))
+            selection_shape.append(selection_slice_size)
+        else:
+            selection_shape.append(size)
+    return tuple(selection_shape)
+
+
+class PartialChunkIterator(object):
```

Comment:
Python 3 only, we can drop objects now :-P
```suggestion
class PartialChunkIterator:
```